opennms-doc/guide-admin/src/asciidoc/text/minion/jms.adoc

```diff
@@ -0,0 +1,81 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+=== Using JMS
+
+By default, _{opennms-product-name}_ uses the _JMS_ protocol with an _ActiveMQ_ broker to communicate with _Minions_.
+This is used for both issuing remote procedure calls (RPCs, ie. ping this host) and for transporting unsolicited messages such as SNMP traps and syslog messages.
+_{opennms-product-name}_ provides an embedded _ActiveMQ_ broker to help simplify installation.
+
+
+TIP: It is also possible (and recommended for large installations) to use an external an external broker.
+
+==== Tuning the ActiveMQ broker
+
+The settings for the embedded _ActiveMQ_ broker are found in `$OPENNMS_HOME/etc/opennms-activemq.xml`.
+Memory and storage limits are conservative by default and should be tuned to accommodate your workload.
+Consider increasing the `memoryUsage` (defaults to 20MB) to `512MB` or greater, assuming you have enough heap available.
+
+TIP: If the memory limit is reached, https://activemq.apache.org/producer-flow-control.html[flow control] will prevent messages
+from being published to the broker.
+
+==== Monitoring the ActiveMQ broker using the Karaf shell
+
+The `opennms-activemq:stats` command available via the _Karaf shell_ can be used to show statistics about the embedded broker:
+
+[source]
+----
+opennms-activemq:stats
+----
+
+TIP: If the command is not available, try installing the feature using `feature:install opennms-activemq-shell`
```

Comment:
I've just installed the opennms-core / opennms-webapp-jetty / opennms-webapp-hawtio packages. It was required to run `feature:install opennms-activemq-shell`. It seems to me so far this step is always required and not just a tip just in case?

Reply:
Required in this branch, but will be part of the default features in `develop`.

Is there an easier way to describe it? (I was trying not to go too deep into feature management here.)

Reply:
when it is a default feature, everything is fine. 👍